### ISSUE TITLE
testing --range-culling parameter

### ISSUE THREAD

**Issue by advaitb** [bbuchfink/diamond#553]
Hi,

I've been stuck on this for a while now and would really appreciate your insight on this. I'm trying to test the `--range-culling` parameter by running DIAMOND (v2.0.11) on a chimeric long read. To simulate this, I've concatenated  a certain number of ORFs (all these are present in the DB) and want to see if I can try and recover all the ORFs along the length fo the read. These are error free so I expect DIAMOND to giuve me a high accuracy. Here are the diamond parameters I'm using:   

> diamond blastx -q chunk \       
        -d dbs/diamond/uniref.mini.dmnd" \             
        -o outchunk \              
        --evalue 10 \             
        --threads 10 \           
        --block-size 5 \                
        --index-chunks 4 \                  
        --salltitles \                 
        --fast \                 
        --frameshift 15 \               
        --range-culling \             
        --min-orf 10 \            
        --masking 0 \             
        -k 0 \               
       --max-hsps 0 \               
        -f 100 

However, I seem to be running into issues recognizing all the different ORFs on the read. Fo example on the read with just one ORF (AAM01497.1 ), I get the correct hit. But when I concatenate it with another ORF from a different organism (AAM01497.1 + NP_465116.1), I still get just 1 hit instead of 2. The problem becomes more complex as more ORFs are appended, where I can only recover an arbitrary number of ORFs at a given time. Moreover which ORF I recover seems to depend on the order of the ORFs. Just as a sanity check ran each ORF separately and saw that I could recover all hits so not sure what's up here.

Is there any way I can make DIAMOND recover all the ORFs? I've tried a bunch of different parameter combinations including different values of `-F`, `--min-orf`, `--sensitive\--very-sensitive`, `-k` as well as `--max-hsps` but nothing seems to work. 

Regards,
Advait

**Comment by bbuchfink** [comment]
Ok thanks for reporting this, I will try to reproduce the problem.

**Comment by advaitb** [comment]
Thanks @bbuchfink, I'm sharing some info and small test files with you that may help with figuring this out. I'm using (a shortened) UniRef100 as the DB. [Here](https://drive.google.com/file/d/1DkfwM3roRyJrnGGhA1TzZHAceoD5b-Gz/view?usp=sharing) are some example sequences and [here ](https://docs.google.com/document/d/12tZVxFWJ_dTUrhJ0EI3t0_RcrU28zB0DAGtrqFFT7KQ/edit?usp=sharing)is the corresponding mapping of their accessions to UniRef100 IDs present in the DB.

**Comment by bbuchfink** [comment]
I can't reproduce the problem using the test query file you provided. Running this:
`diamond blastx -q ~/mini_nc_test.fa -d uniref100.dmnd -o out2 -c1 --log -F 15 -k1 --range-culling -p32`
Seems to give me the correct output:

```
AAM01497.1      UniRef100_Q8TYL6        99.8    430     1       0       1293    4       1       430     0.0     863
AAM01497.1_NP_465116.1  UniRef100_Q8TYL6        99.8    430     1       0       1293    4       1       430     1.79e-310       863
AAM01497.1_NP_465116.1  UniRef100_Q8Y6U1        100     343     0       0       2325    1297    1       343     1.61e-240       682
AAM01497.1_NP_465116.1_NP_308532.1      UniRef100_Q8TYL6        99.8    430     1       0       1293    4       1       430     4.86e-304       863
AAM01497.1_NP_465116.1_NP_308532.1      UniRef100_P69680        100     428     0       0       2326    3609    1       428     2.42e-293       836
AAM01497.1_NP_465116.1_NP_308532.1      UniRef100_Q8Y6U1        100     343     0       0       2325    1297    1       343     1.38e-234       682
```

Please double check this and create another test case if possible.

**Comment by advaitb** [comment]
Hi @bbuchfink ,

Thank you so much for your response. As a part of my pipeline, I had g old script to format .daa to tab/xml  using `view` but with was  the `--top 5` parameter (I missed this) which was causing issues with filtering hits because range-culling now contains hits to bit-scores with a higher variance. As soon as that was fixed, I was able to see all hits. Thanks for looking into this and for the great tool. I look forward to using this mode in my research.
